File: src/main/java/io/github/dqualizer/dqlang/types_new/dam/DomainArchitectureMappingObject.java
package io.github.dqualizer.dqlang.types_new.dam;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.ToString;

import java.util.LinkedHashSet;

@Getter
@ToString
public class DomainArchitectureMappingObject {

    @JsonProperty("dq_id")
    private String dqID;
    private String name;
    @JsonProperty("operation_id")
    private String operationID;
    private String type;
    @JsonProperty("implements")
    private Set<String> implementations;
    private Set<String> objects;
    private Set<Activity> activities;
}

File: src/main/kotlin/dqualizer/dqlang/archive/loadtesttranslator/dqlang/domainarchitecturemapping/DomainArchitectureMappingObject.kt
package dqualizer.dqlang.archive.loadtesttranslator.dqlang.domainarchitecturemapping

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Getter
import lombok.ToString

@Getter
@ToString
class DomainArchitectureMappingObject {
    @JsonProperty("dq_id")
    private val dqID: String? = null
    private val name: String? = null

    @JsonProperty("operation_id")
    private val operationID: String? = null
    private val type: String? = null

    @JsonProperty("implements")
    private val implementations: LinkedHashSet<String>? = null
    private val objects: LinkedHashSet<String>? = null
    private val activities: LinkedHashSet<Activity>? = null
}
